parlai/crowdsourcing/utils/acceptability.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import re
+from typing import List
+
+from parlai.utils.safety import OffensiveStringMatcher
+
+
+class AcceptabilityChecker:
+    def __init__(self):
+        self.offensive_lang_detector = OffensiveStringMatcher()
+
+    def check_messages(
```

Comment:
for reusability purpose, it would be better to separate the each violation checker within the check_messages, or at least pass a flag to disable some checkers. e.g. in the adversarial tasks, we won't block any human messages flagged by the safety string matcher intentionally; similarly for greetings, I would imagine if the human first utterance is greeting unlike the q-function style (fixed opener), it should not be punished.

Reply:
Yes, good point - changed this to add an option to disable certain violations